src/pytest_reportlog/plugin.py

```diff
@@ -67,6 +67,16 @@ def pytest_runtest_logreport(self, report):
         )
         self._write_json_data(data)
 
+    def pytest_warning_recorded(self, warning_message, when, nodeid, location):
+        data = vars(warning_message)
```

Comment:
> it adds new entries that are not serialized pytest objects, so it would need documentation (and the tests would need a dump of the expected values).

Agreed, let's explicitly add the fields we think are useful instead.